### PR TITLE
Add JSON functionality

### PR DESCRIPTION
**Pull request by reecetrodriguez**
Encoding/decoding functionality implemented
between server and client

### CODE REVIEW

File: backend/cmd/backend/main.go [hunk 1]
```diff
@@ -1,21 +1,63 @@
...
+		log.Fatalln("Could not create datastore. Received err:", datastoreErr)
 	}
 	log.Println("Connected to the datastore!")
 
 	log.Println("Starting Amarna backend...")
+
 	for {
+		conn, connErr := l.Accept()
+		if connErr != nil {
+			log.Panicln(connErr)
```
Comment (gqo) [inline backend/cmd/backend/main.go]: Don't use panic. This will kill our backend by calling panic() after it logs.

File: backend/cmd/backend/main.go [hunk 2]
```diff
@@ -1,21 +1,63 @@
...
+	decode := json.NewDecoder(c)
+
+	decodeErr := decode.Decode(&header)
+	if decodeErr != nil {
+		log.Println(decodeErr)
+	}
+
+	fmt.Printf("Action: %s, CompName: %s\n", header.Action, header.CompName)
+
+	c.Close()
```
Comment (gqo) [inline backend/cmd/backend/main.go]: Defer this at the start of the function instead and change the net.Conn parameter to take a pointer

```go
func handleRequest(c *net.Conn) {
    defer c.Close()
    ...
}
```

File: backend/cmd/testfrontend/main.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+		Action:   "GET",
+		CompName: "TEST",
+	}
+
+	jsonEncodeErr := encode.Encode(inputJSON)
+
+	if jsonEncodeErr != nil {
+		fmt.Printf("Ertror is: %v", jsonEncodeErr)
+	}
+	c.Close()
```
Comment (gqo) [inline backend/cmd/testfrontend/main.go]: defer this when you start the connection but after the error check so...

```go
func main() {
    c, e := ...
    if e != nil { ... }
    defer c.Close()
    ...
}
```

File: backend/cmd/testfrontend/tempCodeRunnerFile.go [hunk 4]
```diff
@@ -0,0 +1,4 @@
+type Packet struct {
```
Comment (gqo) [inline backend/cmd/testfrontend/tempCodeRunnerFile.go]: This file seems irrelevant. Delete it.

**Review by gqo — CHANGES_REQUESTED** [review]
Good but small changes.

**Review by gqo — APPROVED** [review]
great

**Review by amvasquez — APPROVED** [review]
up vote
